### PR TITLE
[ZoneMinder] Bugfixes

### PR DESCRIPTION
**Pull request by Mr-Eskildsen**
Fixes the problem when ZoneMinder monitor has no events

### CODE REVIEW

File: addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java [hunk 1]
```diff
@@ -661,7 +668,10 @@ public void updateAvaliabilityStatus(IZoneMinderConnectionInfo connection) {
                     statusDetail = ThingStatusDetail.COMMUNICATION_ERROR;
                     statusDescription = "Session lost connection to ZoneMinder Server";
                     updateBridgeStatus(newStatus, statusDetail, statusDescription);
-                    logger.error("{}: {}", getLogIdentifier(), statusDescription);
+
+                    if (curBridgeStatus != ThingStatus.OFFLINE) {
+                        logger.error("{}: {}", getLogIdentifier(), statusDescription);
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java]: You should actually not log an error at all - the OFFLINE status will already be logged as an event with the statusDescription.

Same for all other occurrences below.

File: addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java [hunk 2]
```diff
@@ -782,14 +816,17 @@ protected void onFetchData() {
                     channelDaemonFrame = false;
                 }
             }
-        } finally {
+        } finally
+
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java]: revert this change

File: addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java [hunk 3]
```diff
@@ -782,14 +816,17 @@ protected void onFetchData() {
...
+        {
             releaseSession();
         }
 
         RecalculateChannelStates();
 
-        if ((channelForceAlarm == false) && (channelAlarmedState == false)
-                && (DataRefreshPriorityEnum.HIGH_PRIORITY == getRefreshPriority())) {
+        if ((channelForceAlarm == false) && (channelAlarmedState == false) && (DataRefreshPriorityEnum.HIGH_PRIORITY ==
+
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java]: revert this change

File: addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java (hunk 2 above)
Comment (Mr-Eskildsen) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java]: By "Revert this change" you mean don't logError? I have changed some logic as well, those changes is relevant in my opinion

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java]: I mean the two useless empty lines at exactly this place here.

Comment (Mr-Eskildsen) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java]: Sorry I misunderstood your first comment "Same for all other occurences below", as if it was the following comments. Reading it more carefully gives fully sense, off course the above change should be reverted :-)

File: addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java (hunk 1 above)
Comment (Mr-Eskildsen) [inline addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java]: Would it be acceptable to do some debug logging instead? Or is all types of logging no-go in this context?

**Comment by Mr-Eskildsen** [comment]
All done, your review gave much more sense when looking at the code :-)
